lib/dialects/mysql/index.js

```diff
@@ -171,27 +171,18 @@ Object.assign(Client_MySQL.prototype, {
 
   canCancelQuery: true,
 
-  cancelQuery(connectionToKill) {
-    const acquiringConn = this.acquireConnection();
-
-    // Error out if we can't acquire connection in time.
-    // Purposely not putting timeout on `KILL QUERY` execution because erroring
-    // early there would release the `connectionToKill` back to the pool with
-    // a `KILL QUERY` command yet to finish.
-    return timeout(acquiringConn, 100)
```

Comment:
Short story: I remove the `100ms` timeout here.

I think the logic was intended to "cancel the cancellation" if it took too long.  In reality, it was just returning control to the caller after 100ms; yet the actual cancellation operation was still running.